bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java

```diff
@@ -0,0 +1,234 @@
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.yioremote.internal.YIOremoteBindingConstants.YIOREMOTEHANDLESTATUS;
+import org.openhab.binding.yioremote.internal.YIOremoteBindingConstants.YIOREMOTEMESSAGETYPE;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link YIOremoteDockHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Michael Loercher - Initial contribution
+ */
+@NonNullByDefault
+public class YIOremoteDockHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(YIOremoteDockHandler.class);
+
+    private @Nullable YIOremoteConfiguration config;
+    private WebSocketClient YIOremote_DockwebSocketClient = new WebSocketClient();
+    private YIOremoteDockWebsocket YIOremoteDockwebSocket_ClientSocket = new YIOremoteDockWebsocket();
+    private ClientUpgradeRequest YIOremote_DockwebSocketClientrequest = new ClientUpgradeRequest();
+    private @Nullable URI URI_yiodockwebsocketaddress;
+    private YIOREMOTEHANDLESTATUS YIOREMOTEHANDLESTATUS_actualstatus = YIOREMOTEHANDLESTATUS.UNINITIALIZED;
+    private @Nullable Future<?> pollingJob;
+    private String send_ircode = "";
+
+    public YIOremoteDockHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing!");
+        config = getConfigAs(YIOremoteConfiguration.class);
+
+        // set the thing status to UNKNOWN temporarily and let the background task decide for the real status.
+        // the framework is then able to reuse the resources from the thing handler initialization.
+        // we set this upfront to reliably check status updates in unit tests.
+        updateStatus(ThingStatus.UNKNOWN);
+        try {
+            logger.debug("Starting generating URI_yiodockwebsocketaddress");
+            URI_yiodockwebsocketaddress = new URI("ws://" + config.yiodockhost + ":946");
+            logger.debug("Finished generating URI_yiodockwebsocketaddress");
+            YIOREMOTEHANDLESTATUS_actualstatus = YIOREMOTEHANDLESTATUS.AUTHENTICATION_PROCESS;
+            try {
+
+                logger.debug("Starting websocket Client");
+                YIOremote_DockwebSocketClient.start();
+                logger.debug("Started websocket Client");
+            } catch (Exception e) {
+                logger.warn("Web socket start failed {}", e.toString());
+                // throw new IOException("Web socket start failed");
+            }
+            try {
+                logger.debug("Connect websocket client");
+                YIOremote_DockwebSocketClient.connect(YIOremoteDockwebSocket_ClientSocket, URI_yiodockwebsocketaddress,
```

Comment:
`initialize()` should return fast. This should be executed in a scheduler task, like in the generated skeleton code.